cpp/benchmarks/hashing/hash.cpp

```diff
@@ -15,47 +15,47 @@
  */
 
 #include <benchmarks/common/generate_input.hpp>
-#include <benchmarks/fixture/benchmark_fixture.hpp>
-#include <benchmarks/synchronization/synchronization.hpp>
 
 #include <cudf/hashing.hpp>
 #include <cudf/table/table.hpp>
 #include <cudf/utilities/default_stream.hpp>
 
-class HashBenchmark : public cudf::benchmark {};
+#include <nvbench/nvbench.cuh>
 
-enum contains_nulls { no_nulls, nulls };
-
-static void BM_hash(benchmark::State& state, cudf::hash_id hid, contains_nulls has_nulls)
+static void bench_hash(nvbench::state& state)
 {
-  cudf::size_type const n_rows{(cudf::size_type)state.range(0)};
-  auto const data = create_random_table({cudf::type_id::INT64}, row_count{n_rows});
-  if (has_nulls == contains_nulls::no_nulls)
+  auto const num_rows  = static_cast<cudf::size_type>(state.get_int64("num_rows"));
+  auto const has_nulls = static_cast<bool>(state.get_int64("nulls"));
+  auto const hash_name = state.get_string("hash_name");
+
+  auto const data =
+    create_random_table({cudf::type_id::INT64, cudf::type_id::STRING}, row_count{num_rows});
+  if (!has_nulls) {
     data->get_column(0).set_null_mask(rmm::device_buffer{}, 0);
+    data->get_column(1).set_null_mask(rmm::device_buffer{}, 0);
+  }
```

Comment:
```suggestion
  data_profile profile;
  if (!has_nulls) {
    profile.set_null_probability(0.0);
  }
  auto const data =
    create_random_table({cudf::type_id::INT64, cudf::type_id::STRING}, row_count{num_rows});
```
or consider using `data_profile_builder`